pkg/operators/register.go

```diff
@@ -36,9 +56,119 @@ func init() {
+// Future improvements on the search and storage roadmap, such as introducing resource federation,
+// should eliminate the need for this workaround. Once global search capabilities are available,
+// they should replace this implementation.
+type unifiedStorageFactory struct {
+	cfg       resource.RemoteResourceClientConfig
+	tracer    trace.Tracer
+	conn      grpc.ClientConnInterface
+	indexConn grpc.ClientConnInterface
+}
+
+func NewUnifiedStorageClientFactory(cfg unifiedStorageConfig, tracer tracing.Tracer) (resources.ResourceStore, error) {
+	registry := prometheus.NewPedanticRegistry()
+	conn, err := unified.GrpcConn(cfg.GrpcAddress, registry)
+	if err != nil {
+		return nil, fmt.Errorf("create unified storage gRPC connection: %w", err)
+	}
+
+	indexAddress := cfg.GrpcIndexAddress
+	if indexAddress == "" {
+		indexAddress = cfg.GrpcAddress
+	}
+
+	indexConn, err := unified.GrpcConn(indexAddress, registry)
+	if err != nil {
+		return nil, fmt.Errorf("create unified storage index gRPC connection: %w", err)
+	}
+
+	return &unifiedStorageFactory{
+		tracer:    tracer,
+		conn:      conn,
+		indexConn: indexConn,
+	}, nil
+}
+
+func (s *unifiedStorageFactory) getClient(ctx context.Context) (resource.ResourceClient, error) {
+	return resource.NewRemoteResourceClient(s.tracer, s.conn, s.indexConn, s.cfg)
+}
+
+func (s *unifiedStorageFactory) CountManagedObjects(ctx context.Context, in *resourcepb.CountManagedObjectsRequest, opts ...grpc.CallOption) (*resourcepb.CountManagedObjectsResponse, error) {
+	client, err := s.getClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("get unified storage client: %w", err)
+	}
+	return client.CountManagedObjects(ctx, in, opts...)
+}
+
+func (s *unifiedStorageFactory) ListManagedObjects(ctx context.Context, in *resourcepb.ListManagedObjectsRequest, opts ...grpc.CallOption) (*resourcepb.ListManagedObjectsResponse, error) {
+	client, err := s.getClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("get unified storage client: %w", err)
+	}
+
+	return client.ListManagedObjects(ctx, in, opts...)
+}
+
+func (s *unifiedStorageFactory) Search(ctx context.Context, in *resourcepb.ResourceSearchRequest, opts ...grpc.CallOption) (*resourcepb.ResourceSearchResponse, error) {
+	client, err := s.getClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("get unified storage client: %w", err)
+	}
+
+	return client.Search(ctx, in, opts...)
+}
+
+func (s *unifiedStorageFactory) GetStats(ctx context.Context, in *resourcepb.ResourceStatsRequest, opts ...grpc.CallOption) (*resourcepb.ResourceStatsResponse, error) {
+	client, err := s.getClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("get unified storage client: %w", err)
+	}
+
+	return client.GetStats(ctx, in, opts...)
+}
+
+type unifiedStorageConfig struct {
+	GrpcAddress      string
+	GrpcIndexAddress string
+	ClientConfig     resource.RemoteResourceClientConfig
 }
 
 func runJobController(opts standalone.BuildInfo, c *cli.Context, cfg *setting.Cfg) error {
```

Comment:
The setup to test this got way too convoluted to start things manually every time:
- Signer
- DB
- Secrets
- Provisioning API Server
- Unified Storage
- Folders API Server
- Dashboards API Server
- Eventually repository controller.

All those things must work at the same time, and be correctly configured to talk to each other.  Any recommended mechanism in how to create a dev environment for this quickly?